jscomp/frontend/ast_derive_projector.ml

```diff
@@ -146,14 +151,19 @@ let init () =
                     {
                       pcd_name = {loc; txt = con_name};
                       pcd_args;
-                      pcd_loc = _;
+                      pcd_loc;
                       pcd_res;
                     }
                   ->
                     let pcd_args =
                       match pcd_args with
                       | Pcstr_tuple pcd_args -> pcd_args
-                      | Pcstr_record _ -> assert false
+                      | Pcstr_record _ ->
```

Comment:
Why does this appear twice?
Are there 2 different ways to trigger the error?
In any case better refactor the error message printing and avoid duplication.

Reply:
Ah yes, sorry I just copy pasted 👍🏼, the one is for structure gen the other is for signature gen. If it hits the case in one it will also in the other so you will only see one message.

Reply:
@cristianoc I've refactored it. See the following commit. Thanks!